news/6396.bugfix

```diff
@@ -0,0 +1,2 @@
+Allow user to pass --prefix="" from command-line to override global prefix settings with an empty
+prefix.
```

Comment:
Why is this needed? Is there an issue that was raised that this change addresses? I'm not at all sure I can think of a reasonable situation where I'd expect anyone to have `--prefix` set in global config - that seems like a bad thing to do in the first place (so I'm suspicious of any situation where this feature would be needed).

Reply:
Hi, yes - Sorry for not being more clear! `--prefix` is (somewhat notoriously) set in a system-global config when installing `pip` with `brew` on MacOS. A common workaround when trying to use `--target` with a `brew`-installed `pip` is to drop a config file into the local directory container `prefix=`.

This change obviates the need for the extra file, and allows this workaround to be achievable via command-line invocation.

Reply:
Ah, OK. I see the need in that case (although personally, I think it should be fixed upstream in Homebrew, not something we should deal with in pip).

Maybe `--target` should ignore `--prefix`? The two seem incompatible to me. It seems like the root issue warrants more design discussion (probably involving some people who, unlike me, are familiar with OSX and Homebrew ;-)).

Reply:
That's fair - I'm no expert in this ecosystem, but my proposed fix (allow empty command-line prefix) seemed like a simple and likely harmless change. I suspect that fixing the larger issue is significantly more involved - I thought this might be a good compromise that could help a number of people. I'd love for input from others more knowledgable about this than myself, though.